src/main/grammars/ElmParser.bnf

```diff
@@ -472,7 +472,18 @@ ListPattern ::=
     LEFT_SQUARE_BRACKET [<<non_empty_list Pattern>>] RIGHT_SQUARE_BRACKET
 
 UnionPattern ::=
-    UpperCaseQID Pattern*
+    UpperCaseQID UnionArgumentPattern*
+
+private UnionArgumentPattern ::=
```

Comment:
Unparenthesized union patterns and `as` clauses can't appear as direct children of a union pattern.